### ISSUE TITLE
federation broken in 1.10: federation typedefs are missing

### ISSUE THREAD

**Issue by bsmedberg-xometry** [dotansimha/graphql-code-generator#3215]
**Describe the bug**
 

**To Reproduce**

A dirt-simple schema that uses federation works in 1.8.3 and is broken in 1.10.

Sandbox repro environment available at https://codesandbox.io/s/serene-faraday-qoech

1. My GraphQL schema:

```graphql
extend type User @key(fields: "userId") {
    userId: String! @external
}

extend type Query {
    currentOnCall: User
}
```

2. My GraphQL operations:
(no operations)

3. My `codegen.yml` config file:

```yml
schema: schema.graphql
config:
  federation: true
  avoidOptionals: true
  immutableTypes: true
  useIndexSignature: true
  typesPrefix: "I"
  noschemaStitching: true
generates:
  types.ts:
    plugins:
      - typescript
      - typescript-resolvers

```

**Expected behavior**
Codegen should work.

**Actual behavior**

```
sandbox@sse-sandbox-qoech:/sandbox$ yarn generate
yarn run v1.19.1
$ yarn codegen
$ graphql-codegen --config ./codegen.yml
  ✔ Parse configuration
  ❯ Generate outputs
    ❯ Generate types.ts
      ✖ Load GraphQL schemas
        → Failed to load schema
        Load GraphQL documents
        Generate


 Found 1 error

  ✖ types.ts
    Failed to load schema from schema.graphql:

            Cannot extend type "User" because it is n
ot defined.

    Unknown directive "key".

    Unknown directive "external".

    Cannot extend type "Query" because it is not defi
ned.

    Unknown type "User".

    Unknown type "Query".
            Error: Cannot extend type "User" because
it is not defined.

    Unknown directive "key".

    Unknown directive "external".

    Cannot extend type "Query" because it is not defi
ned.

    Unknown type "User".

    Unknown type "Query".
        at assertValidSDL (/sandbox/node_modules/grap
hql/validation/validate.js:108:11)
        at Object.buildASTSchema (/sandbox/node_modul
es/graphql/utilities/buildASTSchema.js:71:34)
        at Object.buildSchemaFromTypeDefinitions
(/sandbox/node_modules/@kamilkisiela/graphql-tools/di
st/generate/buildSchemaFromTypeDefinitions.js:23:28)
        at Object.makeExecutableSchema
(/sandbox/node_modules/@kamilkisiela/graphql-tools/di
st/makeExecutableSchema.js:26:29)
        at Object.mergeSchemasAsync
(/sandbox/node_modules/@graphql-toolkit/schema-mergin
g/index.cjs.js:890:25)

            GraphQL Code Generator supports:
              - ES Modules and CommonJS exports (expo
rt as default or named export "schema")
              - Introspection JSON File
              - URL of GraphQL endpoint
              - Multiple files with type definitions
(glob expression)
              - String in config file

            Try to use one of above options and run c
odegen again.
    Error: Failed to load schema
        at loadSchema (/sandbox/node_modules/@graphql
-codegen/cli/bin.js:351:15)
    Error: Failed to load schema
        at loadSchema (/sandbox/node_modules/@graphql
-codegen/cli/bin.js:351:15)


Something went wrong
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Environment:**

- OS: sandbox (Linux)
- `@graphql-codegen/...`: works with 1.8.3, fails with 1.10
- NodeJS: 10.17.0

**Comment by MichielDeMey** [comment]
Currently have the same issue, haven't found a solution yet.

**Comment by ardatan** [comment]
Fixed on this PR;
You can try it with the canary version: `1.10.1-alpha-2faa59c0.4`

**Comment by MichielDeMey** [comment]
I can confirm that the canary version solves the issue. 👍

**Comment by MichielDeMey** [comment]
@ardatan 

One thing I have noticed is that the `Query` property is not properly added to the `Resolvers` type.

When you extend the Query (without an existing Query definition):
```js
const typeDefs = gql`
  extend type Query {
    me: User
  }

  type User @key(fields: "id") {
    ...
  }
`
```

It generates the following `Resolvers` (missing `Query`)

```js
export type Resolvers<ContextType = any> = {
  User?: UserResolvers<ContextType>,
};
```

But when adding the actual `Query` definition (not extending) it does generate the correct `Resolvers` type.

```js
const typeDefs = gql`
  type Query {
    me: User
  }

  type User @key(fields: "id") {
    ...
  }
`
```

```js
export type Resolvers<ContextType = any> = {
  Query?: QueryResolvers<ContextType>,
  User?: UserResolvers<ContextType>,
};
```

According to the Federated Apollo documentation, there is no need to define `Query` or `Mutation` as they are handled by Apollo.

See: https://www.apollographql.com/docs/apollo-server/federation/core-concepts/#root-queries-and-mutations

> There is no need to explicitly define `Query` or `Mutation` base types anywhere; Apollo automatically handles this for you.

Not sure if this is a bug or simply went over everyone's head. 🙂

**Comment by ardatan** [comment]
You're right it should be there!
You can try`1.10.1-alpha-3fc7bb8a.6` :)

**Comment by MichielDeMey** [comment]
Again, that fixes the issue. 🙂👍

**Comment by dotansimha** [comment]
Fixed in 1.11.0
